### DISCUSSION TITLE
How to exclude ValidationError from Schema?

### DISCUSSION THREAD

**Discussion by ticdenis** [fastapi/fastapi#8943] [Questions]
I would like to exclude the ValidationError model from the OpenAPI Schema because I have completely overwritten it like this
```python
from fastapi import FastAPI, Request
from fastapi.exceptions import RequestValidationError
from fastapi.responses import ORJSONResponse
from starlette.exceptions import HTTPException

app = FastAPI(default_response_class=ORJSONResponse)

def exception_handler(req: Request, err: Exception) -> ORJSONResponse:
    # ...

app.add_exception_handler(HTTPException, exception_handler)
app.add_exception_handler(RequestValidationError, exception_handler)
app.add_exception_handler(Exception, exception_handler)

# ...
```
and I can't find any current training without directly updating the JSON schema to do so.

Is there any other way to exclude it?

**Comment by tacozMacleo** [comment]
Normally you raise it, or make a try-except to get hold of it.
Maybe your should add a Minimal working example on what it is you want! or a longer explanation. 
But in case you just want to use the same ValidationError. From the file: `fastapi/exceptions.py` it look like he is using the `ValidationError` from `pydantic`, you can an example on how to use it here: [Validation Usage](https://pydantic-docs.helpmanual.io/usage/validators/).
Here are the example from the site:
```python
from pydantic import BaseModel, ValidationError, validator


class UserModel(BaseModel):
    name: str
    username: str
    password1: str
    password2: str

    @validator('name')
    def name_must_contain_space(cls, v):
        if ' ' not in v:
            raise ValueError('must contain a space')
        return v.title()

    @validator('password2')
    def passwords_match(cls, v, values, **kwargs):
        if 'password1' in values and v != values['password1']:
            raise ValueError('passwords do not match')
        return v

    @validator('username')
    def username_alphanumeric(cls, v):
        assert v.isalnum(), 'must be alphanumeric'
        return v


user = UserModel(
    name='samuel colvin',
    username='scolvin',
    password1='zxcvbn',
    password2='zxcvbn',
)
print(user)
#> name='Samuel Colvin' username='scolvin' password1='zxcvbn' password2='zxcvbn'

try:
    UserModel(
        name='samuel',
        username='scolvin',
        password1='zxcvbn',
        password2='zxcvbn2',
    )
except ValidationError as e:
    print(e)
    """
    2 validation errors for UserModel
    name
      must contain a space (type=value_error)
    password2
      passwords do not match (type=value_error)
    """

```

**Comment by ticdenis** [comment]
I edited the issue description adding more context.
---
I'd like to exclude ValidationError model from OpenAPI Schema generated through FastAPI once I have overwritten the exception and validation handlers.

**Comment by ycd** [comment]
@ticdenis Exactly where do you want to override, There are [two places](https://swagger.io/specification/#components-object) that shows your schema, one in responses, and another one in the components/schema. Or do you want to exclude both?

**Comment by ticdenis** [comment]
@ycd I want exclude the Schema "ValidationError".

**Comment by Kludex** [comment]
This is duplicated issue. This should solve your problem: https://github.com/tiangolo/fastapi/issues/660.

**Comment by ThomasThelen** [comment]
I attempted the linked solution above but failed to get it working for POST requests. Maybe someone else that runs across this issue will have had a similar experience.
